macchiato.py

```diff
@@ -2,102 +2,168 @@
 import sys
 import tempfile
 import tokenize
+from typing import IO, Iterable, List, NamedTuple, Optional, Tuple, cast
 
 import black
 
 
 __version__ = "1.2.0"
 
+_single_indent = " " * 4
 
-def macchiato(in_fp, out_fp, args=None):
-    if args is None:
-        args = []
 
-    # Read input.
-    lines = in_fp.readlines()
+class WrapInfo(NamedTuple):
```

Comment:
can't say i like namedtuples in general, but i guess using `dataclass` means this code won't work with python3.6 anymore...